planning/behavior_path_planner/README.md

```diff
@@ -253,39 +253,118 @@ The Pull Out module is activated when ego-vehicle is stationary and footprint of
 
-![pull_out](./image/pull_out_fig1.drawio.svg)
+![pull_out_collision_check](./image/pull_out_collision_check.drawio.svg)
 
 #### **Path Generation**
 
-The path is generated with a certain margin from the left edge of the shoulder lane.
+There are two path generation methods.
+
+##### **shift pull out**
+
+Pull out distance is calculated by the speed, lateral deviation, and the lateral jerk. The lateral jerk is searched for among the predetermined minimum and maximum values, and the one satisfies ready conditions described above is output.
+
+- Generate the shoulder lane centerline and shift it to the current position.
+- In the section between merge start and end, path is shifted by a method that is used to generate avoidance path (four segmental constant jerk polynomials)
+- Combine this path with center line of road lane
+
+![shift_pull_out](./image/shift_pull_out.drawio.svg)
+
+[Video of how shift pull out works](https://user-images.githubusercontent.com/39142679/181024453-8bb13b8c-d74e-4bdf-b068-cea8215e1258.mp4)
+
+###### parameters for shift pull out
+
+| Name                              | Unit   | Type   | Description                                                                                                          | Default value |
+| :-------------------------------- | :----- | :----- | :------------------------------------------------------------------------------------------------------------------- | :------------ |
+| enable_shift_pull_out             | -      | bool   | flag whether to enable shift pull out                                                                                | true          |
+| shift_pull_out_velocity           | [m/s]  | double | velocity of shift pull out                                                                                           | 2.0           |
+| pull_out_sampling_num             | -      | int    | Number of samplings in the minimum to maximum range of lateral_jerk                                                  | 4             |
+| before_pull_out_straight_distance | [m]    | double | distance before pull out start point                                                                                 | 5.0           |
+| maximum_lateral_jerk              | [m/s3] | double | maximum lateral jerk                                                                                                 | 2.0           |
+| minimum_lateral_jerk              | [m/s3] | double | minimum lateral jerk                                                                                                 | 0.5           |
+| minimum_shift_pull_out_distance   | [m]    | double | minimum shift pull out distance. if calculated pull out distance is shorter than this, use this for path generation. | 20.0          |
+
+##### **geometric pull out**
+
+Generate two arc paths with discontinuous curvature. ego-vehicle stops once in the middle of the path to control the steer on the spot.
```

Comment:
```suggestion
Generate two arc paths with discontinuous curvature. Ego-vehicle stops once in the middle of the path to control the steer on the spot.
```

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/1587/commits/3fe45f3a0539707e19b52304e43c03d3a05e6092